### ISSUE TITLE
First result of tests: errors, improving possibilities and settings.

### ISSUE THREAD

**Issue by Giorgioum** [CompImg/LST-AI#26]
I'm running some tests to see how, and how well, LST-AI works.

1) Again I've got this error ` END OF LEVEL    2
....
Limiting the number of threads to 20
 /home/giorgio/anaconda3/envs/lst_env/bin/hd-bet:4: DeprecationWarning: pkg_resources is deprecated as an API. See https://setuptools.pypa.io/en/latest/pkg_resources.html
  __import__('pkg_resources').require('HD-BET==1.0')` 
that I thought to be fixed by `pip install setuptools==60`. I don't know why this happen, if i run `conda list` I could see `setuptools                60.0.0                   pypi_0    pypi`.

2) ` IFLAG= -1  LINE SEARCH FAILED. SEE DOCUMENTATION OF ROUTINE MCSRCH ERROR RETURN OF LINE SEARCH: INFO= 3 POSSIBLE CAUSES: FUNCTION OR GRADIENT ARE INCORRECT OR INCORRECT TOLERANCESvnl_lbfgs: Error. Netlib routine lbfgs failed.
END OF LEVEL   0
`
Can I repair this error? Does it affect results? 

3) Is there any possibility to improve LST-AI "feeding" it with manual segmentation masks or other ways ? I've tons of cases with T2 hyperintensities lesions, can I use my previous data somehow? Or, in general are there other settings than `--threshold`? Because I see lots of small lesions unidentified and some normal structures (e.g. choroide plexus of IV ventricle or little normal venous formation) segmented.

4) The "other (useful) settings" have to be written after `lst --t1 t1.nii.gz --flair flair.nii.gz --output /mnt/data/lst/results` ?

**Comment by jqmcginnis** [comment]
Hey @Giorgioum,

I will try to address your questions :slightly_smiling_face: 

(1) It's really hard for us to debug and assess the question/problem, if you only partially share the error message, i.e. I would need the whole output to check where the first error happened and how it propagates. 

You can check one thing first - does HD-BET work at all? Try to skull-strip a brain using the hd-bet command line tool. Please refer to the documentation of HD-BET [here](https://github.com/MIC-DKFZ/HD-BET). 

Two more things:
- I would strongly advise you to use a venv environment, and not a conda environment to setup our tool.
- if END OF LEVEL 2 corresponds to greedy's output it is not an error, but merely an indication that step 2 of the optimization process of greedy has finished its execution :slightly_smiling_face: 

(2) No it does not affect the results and you cannot/do not need to repair it.

(3) To fine-tune your model, you would need to retrain the U-Nets in LST-AI. This is definitely feasible, and we encourage this, as we open source everything, but this requires in-depth machine learning knowledge to do. If you want me to point you to papers I can do so, but unfortunately I would not have the time to support the coding. If you think this is ultimately relevant for your task, and you are not happy with LST-AI's performance, perhaps you can get a student to do this (assuming you are in academia?). 
Can you let us know how your scans were acquired? 1.5T/3T? Iso? E.g. our training set consists of istropic FLAIR images, i.e. flow-artefacts are new to the netwerk and may be segmented. You can actually experiment with the threshold (c.f. readme), but apart from that we do not have any other settings. Are your brains skull-stripped already? If so, please adapt your command line instructions :slightly_smiling_face: 

(4) exactly, and if you run `lst --help ` you will get a list of available options and a short explanation on how to use them :slightly_smiling_face:

**Comment by Giorgioum** [comment]
Sorry for the late reply, first of all when I do another test I will send you the exact output, but visually I can assure that BET works fine. Regarding point (3), it would be great if you could point out the documents and sources to do the retrain 🙂. The scans are only Philps 3T isovolumetric (3D) FLAIR and T1 GRE.
Thanks for clarifying point (4).

**Comment by jqmcginnis** [comment]
Sorry for the late reply. I have linked two papers below that show that transfer-learning and fine-tuning may improve the segmentation performance on OOO data. 

https://www.sciencedirect.com/science/article/pii/S0933365721000713
https://pmc.ncbi.nlm.nih.gov/articles/PMC8003655/

Please re-open if you have any more questions.
